### DISCUSSION TITLE
🚀 Cache 'Show Tables' query?

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3341] [Feature Requests]
### Suggest an idea for ExpressionEngine

There are a lot of 'Show Tables FROM db_name', which runs when a 'tables exist' call happens.  And it happens a lot on many update/delete operations. 

It's not a resource hog, but it might be low hanging fruit to cache it or otherwise reduce it.  If you check query logs, there can be a lot in there!

**Comment by intoeetive** [comment]
This seems to be happening on every CP load. Which is IMO redundancy. This seems to be called from the checks we make when loading config, so need to look into ways to optimize it (there is a check for 'config' table existance which I think we only need during upgrades)

There are also 'SHOW COLUMNS` queries on the front-end. Which I think can be absolutely safely cached, as the DB structure is not getting updated from the front-end
